### DISCUSSION TITLE
Recommended development setup?

### DISCUSSION THREAD

**Discussion by rubanooo** [trpc/trpc#1362] [Q&A] [answered]
My setup is as follows: I have a turborepo monorepo with a react frontend under `apps/web` and an trpc express server under `apps/api` 

What tool should I use to generate the trpc types while also running a dev server for the api?

**Comment by sachinraja** [comment, marked as answer]
Do you have a link to the repo? You might not need to generate types.

**Reply by rubanooo** [reply]
I see what you mean. When I import 

```ts
import type { AppRouter } from "api/src/index";
```

instead of:

```ts
import type { AppRouter } from "api";
```

The types work without building. Is this the way it's meant to be used?

**Reply by sachinraja** [reply]
Is `api` symlinked to `node_modules`? You should be able to just add a `"types"` field in your package.json and point it at the source type file, which should allow you to import it from `api` instead of `api/src/index`.

**Reply by rubanooo** [reply]
Awesome! I did not know you could point the `"types"` field to a source type file. I am now able to import the types directly from `"api"` like so:

```ts
import type { AppRouter } from "api";
```

Maybe we can add this to the docs?

**Reply by sachinraja** [reply]
Might be good to add a section on monorepos. There's also a good example of one at https://github.com/KATT/zart which uses the same method that you did but without `"types"`.

**Reply by albingroen** [reply]
I only seem to get type safety for the name of query/mutation, and not for the input, or output. Why is this?

![image](https://user-images.githubusercontent.com/19674362/147964040-0636e68b-b526-4f67-b5ba-0783804730c9.png)

![image](https://user-images.githubusercontent.com/19674362/147963977-105f2b69-5d6c-4f52-979c-5dbf4393cc60.png)

**Reply by albingroen** [reply]
If I install `@trpc/server` into my client I seem to get the correct input and output types.

**Reply by sachinraja** [reply]
I would think that it works without that if you have it installed as a dependency in the server package, but maybe not.

**Reply by albingroen** [reply]
Doesn't seem like it actually
